packages/runtime/container-runtime/src/opLifecycle/opDecompressor.ts

```diff
@@ -43,25 +43,25 @@ export class OpDecompressor {
             const intoString = Uint8ArrayToString(decompressedMessage);
             const asObj = JSON.parse(intoString);
             this.rootMessageContents = asObj;
-
-            return { ...message, contents: this.rootMessageContents[this.processedCount++] };
+            message.contents = this.rootMessageContents[this.processedCount++];
```

Comment:
I'm not sure this is correct. It minimum, this should remove compression metadata from the op (even if nobody cares about it down the line). But overall, we should prefer cloning (i.e. treating args as Readonly<>) over patching. Patching opens up the risk that same payload will be used by someone else (like another container, in case of socket-multi-plexing) and will fail

Reply:
> I'm not sure this is correct.

are you referring to my change or how it works currently?

I had another approach in one of my prototypes in which these classes implement an interface with a `processRemoteMessage` function which returned a (new) message + result, but the code got too messy. I will revisit that. We do need to know whether or not some of these invocations succeeded so we can safely unpack depending on their result (this will be more obvious in the next PR).

Reply:
I will revert this change as it's not really needed right now. When chunking will be added, I will revisit this with a better return type for these methods.